test/helpers/setup.js

```diff
@@ -12,3 +12,32 @@ global.assert = chai.assert;
 chai.use(chaiAsPromised);
 chai.use(chaiThings);
 chai.use(sinonChai);
+
+function paddedHex(intVal) {
+  var s = intVal.toString(16);
+  if (s.length == 1) {
+    s = "0" + s;
+  }
+  return s;
+}
+
+global.extractColors = function(image) {
+  var pixels = image.frames[0].data;
+  var colorMap = new Map();
```

Comment:
Or would it be possible to utilize some existing library for reading pixels, e.g. https://www.npmjs.com/package/get-pixels?

In any case, the code here is obscure enough to require some explanatory comments 😉